src/prog/prog_ProgItem.cpp

```diff
@@ -43,7 +43,7 @@ ProgItem::~ProgItem(void) {
  */
 ProgItem *ProgItem::next(void) const {
 	ProgItem *result = (ProgItem *)inhstruct::DLNode::next(); 
-	if(!result || result->atEnd())
+	if(result && result->atEnd())
```

Comment:
This returns 0 when the next node exists, which is the opposite of the intended behaviour, and probably ends every analysis early. I suppose this is how you end up with no memory leaks...

Reply:
Good catch.

>  I suppose this is how you end up with no memory leaks...
No, with or without this change, we end up with no memory leak.

As described above, in this PR, I add an extra check on result pointer before calling `atEnd()` or `atBegin()` method.